avro-builder/builder/src/main/java/com/linkedin/avroutil1/builder/operations/codegen/util/AvscFileFinderUtil.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2023 LinkedIn Corp.
+ * Licensed under the BSD 2-Clause License (the "License").
+ * See License in the project root for license information.
+ */
+
+package com.linkedin.avroutil1.builder.operations.codegen.util;
+
+import com.linkedin.avroutil1.builder.BuilderConsts;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Enumeration;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.jar.JarFile;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+import org.apache.commons.io.FileUtils;
+
+
+public class AvscFileFinderUtil {
+  private static final String[] EXTENSIONS = new String[]{BuilderConsts.AVSC_EXTENSION};
+  private static final String TEMP_DIR_NAME = "tempCompressedAvroSrc";
+
+  private AvscFileFinderUtil() {
+  }
+
+  public static Collection<File> findFiles(File inputRoot) throws IOException {
```

Comment:
worth documenting. the fact that this will fish inside zips and jars anywhere under the given input root is not immediately obvious and intuitive

Reply:
also - what would this do with dupes? same file (contents) inside and outside of a zip?

Reply:
I'll document but this "fishing" behavior should be as expected: if you pass in a jar/zip by name as an input, I would expect that it looks inside that jar/zip. 

What this method won't do: recurse through your input root and look for jars/zips and get all avsc files inside of those zips/jars.

Reply:
Dupe handling happens after this. avscs found in zips should be treated as any other avsc and how it is handled will depend on your configured dupe behavior